<launch>

  <!--  ************** Global Parameters ***************  -->
  <param name="/use_sim_time" value="true"/>
  <arg name="gui" default="true"/>  


  <!-- Initial pose -->
  <arg name="initial_pose_x" default="1.0"/>
  <arg name="initial_pose_y" default="1.0"/>
  <arg name="initial_pose_a" default="0.13"/>

  <!-- Filename -->
  <arg name="log_filename" default="$(find move_base_benchmark)/launch/log.txt" />
  <arg name="map_filename" default="$(find move_base_benchmark)/maps/shopping_mall/map.yaml" />
  <arg name="world_filename" default="$(find gazebo_world)/world/shopping_mall.world" />
  <arg name="global_frame" value="map" />

  <!--  ************** Navigation ***************  -->
  <node pkg="move_base_benchmark" type="move_base_benchmark" respawn="false" name="move_base_benchmark" output="screen">
    <rosparam file="$(find move_base_benchmark)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find move_base_benchmark)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find move_base_benchmark)/params/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find move_base_benchmark)/params/local_costmap_params_osqp.yaml" command="load" />
    <param name="global_costmap/global_frame" value="$(arg global_frame)" />
    <param name="local_costmap/global_frame" value="$(arg global_frame)" />

    <param name="base_global_planner" value="lattice_path_planner/LatticePathPlannerROS" />
    <param name="LatticePathPlannerROS/primitive_filename" value="$(find lattice_path_planner)/mprim/pr2_5cm.mprim" />
    <param name="LatticePathPlannerROS/do_smoothing" value="true" />
    <param name="LatticePathPlannerROS/smoother_type" value="Ceres" />
    <param name="LatticePathPlannerROS/use_inflation_layer" value="true" />
    <param name="LatticePathPlannerROS/max_plan_length" value="3.0" />
    <param name="planner_frequency" value="1.0" />
    <param name="planner_patience" value="5.0" />


    <param name="base_local_planner" value="ceres_path_smoother/CeresLocalPlannerROS" />
    <rosparam file="$(find move_base_benchmark)/params/teb_tracking_params_ceres.yaml" command="load" />
    <param name="CeresLocalPlannerROS/primitive_filename" value="$(find lattice_path_planner)/mprim/pr2_5cm.mprim" />
    <param name="controller_frequency" value="5.0" />
    <param name="controller_patience" value="15.0" />

    <param name="log_filename" value="$(arg log_filename)" />
    <remap from="cmd_vel" to="RosAria/cmd_vel"/>
    <remap from="odom" to="RosAria/odom"/>
  </node>

  <!--  ****** Maps *****  -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_filename)" output="screen">
    <param name="frame_id" value="$(arg global_frame)"/>
  </node>

  <!--  ****** Localization *****  -->
  <node name="fake_localization" pkg="fake_localization" type="fake_localization" respawn="false" >
    <param name="global_frame_id" value="$(arg global_frame)" />
  </node>
      <!-- Pedsim Gazebo -->
  


  <!--  ****** Start Gazebo and launch world model *****  -->
  <include file="$(find gazebo_world)/launch/world_launch.launch">
    <arg name="world_name" value="$(arg world_filename)"/>
    <arg name="gui" value="$(arg gui)" />
  </include>

  <!-- Spawn p3dx mobile robot -->
  <include file="$(find p3dx_gazebo)/launch/spawn_p3dx.launch">
    <arg name="initial_pose_x" value="$(arg initial_pose_x)" />
    <arg name="initial_pose_y" value="$(arg initial_pose_y)" />
    <arg name="initial_pose_a" value="$(arg initial_pose_a)" />
  </include>
  <!-- ****** Goal Visualization ***** -->
  <node name="goal_visualizer" pkg="move_base_benchmark" type="goal_visualize.py" respawn="false" output="screen">
      <remap from="current_goal" to="/move_base_benchmark/current_goal"/>
  </node>

  <!--  **************** Visualisation ****************  -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find move_base_benchmark)/launch/rviz_navigation.rviz"/>
  


  <!-- TEB Monitor -->
  <!-- <node name="teb_monitor" pkg="robot_gazebo" type="teb_monitor.py" output="screen">
    <param name="num_episodes" value="50" />
  </node>
 -->

</launch>